_posts/2025-05-09-improve-opensearch-cluster-performance.md

```diff
@@ -0,0 +1,180 @@
+
+- The remote store stores segment files and transaction logs. All replicas—including the write replica and search replicas download segments from the remote store.
+- The primary shard writes both segments and transaction logs to the remote store.
+- The write replica downloads segments after the primary completes writing.
+- The search replicas continuously poll the remote store for new segments.
+
+![OpenSearch cluster architecture for search and indexing separation](/assets/media/blog-images/2025-05-09-search-index-separation/rw-separation-architecture.png)
+
+## Benefits 
+
+Separating indexing and search workloads provides the following benefits:
+
+* **Parallel and isolated processing**: Improve throughput and predictability by isolating indexing and search.
+* **Independent scalability**: Scale indexing and search independently by adding `data` or `search` nodes.
+* **Failure resilience**:  Indexing and search failures are isolated, improving availability.
+* **Cost efficiency and performance**: Use specialized hardware—compute-optimized for indexing, memory-optimized for search.
+* **Tuning flexibility**: Optimize performance settings like buffers and caches independently for indexing and search.
+* **Failure isolation**: Indexing and search issues are separated, which helps troubleshoot failures
+* **Scalable design**: Scale indexing and search independently of each other
+
+## Enabling indexing and search separation
+
+For detailed instructions, see [Separate index and search workloads](https://docs.opensearch.org/docs/latest/search-plugins/index-management/index-and-search-separation/).
+
+## Scale to zero with reader/writer separation: The search-only mode
+
+In write-once, read-many scenarios—like log analytics or frozen time-series data—you can reduce resource use after indexing completes. OpenSearch now supports a **search-only mode** through the `_scale` API, allowing you to disable primary shards and write replicas, leaving only search replicas active. 
+
+This significantly reduces storage and compute costs while maintaining full search capabilities and provides the following benefits:
+
+- Scale down indexing capacity when writes are no longer needed
+- Free up disk and memory by removing unnecessary write paths
+- Work smoothly with index lifecycle controls through `_open`/`_close` operations
+- Keep only search replicas active to reduce resource usage
+- Maintain green cluster health with active search paths
+- Auto-recover search replicas without manual intervention
+- Scale back up at any time to resume indexing
+- Scale search replicas based on search traffic during search-only mode
+
+When `search-only` mode is active, OpenSearch maintains cluster health as GREEN when search-only mode is active despite disabling primaries and write replicas, This is because only search replicas are expected to be allocated, and they are all successfully assigned. This ensures operational stability and full search availability.
```

Comment:
```suggestion
OpenSearch maintains cluster health as GREEN when search-only mode is active despite disabling primaries and write replicas. This is because only search replicas are expected to be allocated, and they are all successfully assigned. This ensures operational stability and full search availability.
```